src/hvmc_net/pre_reduce.rs

```diff
@@ -33,8 +33,7 @@ pub fn pre_reduce_run_net(nam: &str, net: &mut hvmc::run::Net) -> Result<(), Str
 
   // Separate the derefing interactions, reduce all the others.
   // Also, always put the Ref on the left side (first element).
-  // The mem::replace thing implements a queue using 2 vecs.
-  let mut rdexes = std::mem::take(&mut net.rdex);
+  let mut rdexes: Vec<_> = net.rdex.drain(|n| n).map(Iterator::collect).unwrap_or(vec![]);
```

Comment:
It would be slightly more idiomatic to use `.unwrap_or_default()` :)
(Also below.)